extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IatpDefaultServicesExtension.java

```diff
@@ -14,27 +14,68 @@
 
 package org.eclipse.edc.iam.identitytrust.core;
 
+import org.eclipse.edc.iam.identitytrust.sts.embedded.EmbeddedSecureTokenService;
 import org.eclipse.edc.identitytrust.SecureTokenService;
+import org.eclipse.edc.jwt.TokenGenerationServiceImpl;
 import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
 import org.eclipse.edc.runtime.metamodel.annotation.Provider;
+import org.eclipse.edc.runtime.metamodel.annotation.Setting;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.security.KeyPairFactory;
 import org.eclipse.edc.spi.system.ServiceExtension;
 import org.eclipse.edc.spi.system.ServiceExtensionContext;
 
+import java.security.KeyPair;
+import java.time.Clock;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
 @Extension("Identity And Trust Extension to register default services")
 public class IatpDefaultServicesExtension implements ServiceExtension {
 
+    @Setting(value = "Alias of private key used for signing tokens, retrieved from private key resolver", defaultValue = "A random EC private key")
+    public static final String STS_PRIVATE_KEY_ALIAS = "edc.iam.sts.privatekey.alias";
+    @Setting(value = "Alias of public key used for verifying the tokens, retrieved from the vault", defaultValue = "A random EC public key")
+    public static final String STS_PUBLIC_KEY_ALIAS = "edc.iam.sts.publickey.alias";
     // not a setting, it's defined in Oauth2ServiceExtension
     private static final String OAUTH_TOKENURL_PROPERTY = "edc.oauth.token.url";
+    @Setting(value = "Self-issued ID Token expiration in minutes. By default is 5 minutes", defaultValue = "" + IatpDefaultServicesExtension.DEFAULT_STS_TOKEN_EXPIRATION)
+    private static final String STS_TOKEN_EXPIRATION = "edc.iam.sts.token.expiration"; // in minutes
+
+    private static final int DEFAULT_STS_TOKEN_EXPIRATION = 5;
+    
+    @Inject
+    private KeyPairFactory keyPairFactory;
+
+    @Inject
+    private Clock clock;
 
     @Provider(isDefault = true)
     public SecureTokenService createDefaultTokenService(ServiceExtensionContext context) {
         context.getMonitor().info("Using the Embedded STS client, as no other implementation was provided.");
+        var keyPair = keyPairFromConfig(context);
+        var tokenExpiration = context.getSetting(STS_TOKEN_EXPIRATION, DEFAULT_STS_TOKEN_EXPIRATION);
```

Comment:
Not sure it is correct to have injected fields within a `@Provider(isDefault = true)` method. Might be ok tough as the injected services are provided by Core extension... @paullatzelsperger WDYT?

Reply:
i think that is correct, as there will be another implementation later (#3499), where  instead of calling a java class, the call goes out over HTTP/REST to a standalone STS. The embedded one would be the "fallback" in that case, or when an `edc.oauth.token.url` is not specified.

Reply:
My comment was more about dependency injection in order to be sure the injected fields are properly initialized before the default provider method is invoked :)